bundles/aws-bundle/src/main/java/org/apache/gravitino/s3/credential/S3TokenProvider.java

```diff
@@ -104,6 +104,10 @@ private StsClient createStsClient(S3CredentialConfig s3CredentialConfig) {
     if (StringUtils.isNotBlank(region)) {
       builder.region(Region.of(region));
     }
+    String stsEndpoint = s3CredentialConfig.stsEndpoint();
+    if (StringUtils.isNotBlank(stsEndpoint)) {
+      builder.endpointOverride(URI.create(stsEndpoint));
```

Comment:
Yes, the endpoint must be a valid URI to work. This check ensures a custom value is assigned when specified by the user.

Reply:
I mean, we'd better check if it is a valid URI before calling `URI.create`,
or we need to catch an exception here?

Reply:
It's hard to maintain how to check a valid URI, it's the responsibility of `URI.create`.  It will throw `IllegalArgumentException` if parse URI failed, should we handle this exception? seems we could do nothing but throw this exception again.

Reply:
Okay. If there is no known way to validate the URI string, can we confirm URI.create() will reject an empty string? The check here is incomplete, that worries me.
Either we do a thorough validation, or we delegate that to URI.create(), both are acceptable. I'm concerned at this kind of partial validation.
If the decision is to delegate the checking to URI.create(), we need to get prepared
for the exception, somewhere along the call chain.

Reply:
It's hard to do a complete check in Gravitino, If user provide a wrong endpoint, the Iceberg catalog wrapper will failed initialize  when creating corresponding credential provider.  Please refer to 
 https://github.com/apache/gravitino/blob/main/iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/IcebergCatalogWrapperManager.java#L115

Reply:
I tend to agree. A thorough check can/should be done in a utils package.
So the conclusion is that we are leaning towards to a trial&error approach.
Let the `URI.create()` try and we are prepared to handle the exception.

Two things left then (as commented previously):

- Shall we check the blank string case? If `URI.create()` silently accepts an empty
  string and we hate it, we should add an additional check here. Preferably adding a
  comment `// URI.create sucks, it cannot detect empty string. This is a stricter check`.
- How we are handling the exception? When something bad happens in this call,
  we cannot just simply leave it there. There could be previous changes we need to
  roll back (e.g. line 105, assuming it has a persistent impact).
  My personal favor is to kill and handle the exception wherever it raises.
  Propagating the exception up the call chain has always been a nightmare to me.

Reply:
if user doesn't config sts endpoint, `URI.create` will throw an exception.

Reply:
yes, it just checks whether `stsEndpoint` is empty or not.

Reply:
So technically I don't need to change any thing , right ?

Reply:
Let's assume stsEndpoint is empty now,

- if URI.create(stsEndpoint) is okay with an empty string, why do we care?
- if URI.create(stsEndpoint) rejects it (throws an exception), why do we check?

This is not about this specific statement. It is about logical thinking.

Reply:
If stsEndpoint is empty, there are two possibilities, 1. user doesn't set it explicitly.  2. user set an empty string.
How should we handle it? The current logic is don't set sts endpoint for sts client.  it's buggy to throw an exception here, WDYT?

Reply:
Okay. If the assumption is that when a user sets an empty string, we treat it as not specified. I'm fine with the current logic.
Please add a comment to this logic.
Thanks for bearing with my noise.

Reply:
@hienduyph could you add some comments here?